presto-main/src/main/java/com/facebook/presto/sql/rewrite/MaterializedViewOptimizationRewriteUtils.java

```diff
@@ -65,13 +83,20 @@ private static Query getQueryWithMaterializedViewOptimization(
             Session session,
             SqlParser sqlParser,
             AccessControl accessControl,
+            WarningCollector warningCollector,
             Query statement,
             QualifiedObjectName materializedViewQualifiedObjectName)
     {
-        ConnectorMaterializedViewDefinition materializedView = metadata.getMaterializedView(session, materializedViewQualifiedObjectName).get();
-        Table materializedViewTable = new Table(QualifiedName.of(materializedView.getTable()));
+        try {
+            ConnectorMaterializedViewDefinition materializedView = metadata.getMaterializedView(session, materializedViewQualifiedObjectName).get();
+            Table materializedViewTable = new Table(QualifiedName.of(materializedView.getTable()));
 
-        Query materializedViewDefinition = (Query) sqlParser.createStatement(materializedView.getOriginalSql());
-        return (Query) new MaterializedViewQueryOptimizer(metadata, session, sqlParser, accessControl, new RowExpressionDomainTranslator(metadata), materializedViewTable, materializedViewDefinition).rewrite(statement);
+            Query materializedViewDefinition = (Query) sqlParser.createStatement(materializedView.getOriginalSql(), createParsingOptions(session, warningCollector));
```

Comment:
It's ok not to pass the warning collector here. We won't generate warnings for generated sql queries.